src/Razor/src/Microsoft.VisualStudio.Editor.Razor/Documents/EditorDocumentManagerListener.cs

```diff
@@ -105,46 +105,44 @@ private async Task ProjectManager_ChangedAsync(ProjectChangeEventArgs e, Cancell
+                            return;
+                        }
 
-                            // GetOrCreateDocument needs to be run on the UI thread
-                            await _joinableTaskContext.Factory.SwitchToMainThreadAsync();
+                        var key = new DocumentKey(e.ProjectFilePath, e.DocumentFilePath);
 
-                            var document = DocumentManager.GetOrCreateDocument(
-                                key, _onChangedOnDisk, _onChangedInEditor, _onOpened, _onClosed);
-                            if (document.IsOpenInEditor)
-                            {
-                                _onOpened(document, EventArgs.Empty);
-                            }
+                        // GetOrCreateDocument needs to be run on the UI thread
+                        await _joinableTaskContext.Factory.SwitchToMainThreadAsync(cancellationToken);
 
-                            break;
-                        }
-
-                    case ProjectChangeKind.DocumentRemoved:
+                        var document = DocumentManager.GetOrCreateDocument(
+                            key, _onChangedOnDisk, _onChangedInEditor, _onOpened, _onClosed);
+                        if (document.IsOpenInEditor)
                         {
-                            // Need to run this even if the solution is closing because document dispose cleans up file watchers etc.
+                            _onOpened(document, EventArgs.Empty);
+                        }
 
-                            // TryGetDocument and Dispose need to be run on the UI thread
-                            await _joinableTaskContext.Factory.SwitchToMainThreadAsync();
+                        break;
+                    }
 
-                            var documentFound = DocumentManager.TryGetDocument(
-                                new DocumentKey(e.ProjectFilePath, e.DocumentFilePath), out var document);
+                    case ProjectChangeKind.DocumentRemoved:
+                    {
+                        // Need to run this even if the solution is closing because document dispose cleans up file watchers etc.
 
-                            // This class 'owns' the document entry so it's safe for us to dispose it.
-                            if (documentFound)
-                            {
-                                document.Dispose();
-                            }
+                        // TryGetDocument and Dispose need to be run on the UI thread
+                        await _joinableTaskContext.Factory.SwitchToMainThreadAsync(cancellationToken);
 
-                            break;
+                        // This class 'owns' the document entry so it's safe for us to dispose it.
+                        if (DocumentManager.TryGetDocument(
+                            new DocumentKey(e.ProjectFilePath, e.DocumentFilePath), out var document))
```

Comment:
I'd revert this too. The comment isn't as clear, and as mentioned before, this style of "capture to local" is very prevalent in Razor Tooling code.

Reply:
`Roslyn doesn't track the relation between separate variables.` © [Cyrus 2022](https://github.com/dotnet/roslyn/issues/64117#issuecomment-1251171203)

In other words, I _have to_ inline a boolean flag here bacause compiler is not smart enough and will complain if I don't do that

Reply:
ohh i see, yes, it would complain about `document`. I missed that.

Reply:
Actually suppression on the Dispose would have been okay, in this case, since it would immediately throw a NRE, so isn't hiding anything, but its fine to leave it too.